platforms/corda-ent/charts/generate-pki/templates/job.yaml

```diff
@@ -78,41 +76,44 @@ spec:
           args:
           - |-
             #!/usr/bin/env sh
-
-            OUTPUT_PATH=${BASE_DIR}
+            validateVaultResponse () {
+              if echo ${2} | grep "errors"; then
+                 echo "ERROR: unable to retrieve ${1}: ${2}"
+                 exit 1
+              fi
+            }
 
             # setting up env to get secrets from vault
             KUBE_SA_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
             VAULT_TOKEN=$(curl -sS --request POST ${VAULT_ADDR}/v1/auth/${KUBERNETES_AUTH_PATH}/login -H "Content-Type: application/json" -d '{"role":"vault-role","jwt":"'"${KUBE_SA_TOKEN}"'"}' | jq -r 'if .errors then . else .auth.client_token end')
-
-            # get the root token from the vault, to upload secrets on vault
-            LOOKUP_PWD_RESPONSE=$(curl -sS --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR}/v1/${RT_SECRET_PREFIX} | jq -r 'if .errors then . else . end')
-            ROOT_TOKEN=$(echo ${LOOKUP_PWD_RESPONSE} | jq -r '.data["rootToken"]')
-
-            # perform check if crl certificates are ready or not, and upload certificate into vault when ready
+            validateVaultResponse 'vault login token' "${VAULT_TOKEN}"
+            
+            cd ${BASE_DIR}/DATA
+            # putting certificate for cenm root 
             COUNTER=1
-            cd ${BASE_DIR}/DATA/crl-files
             while [ "$COUNTER" -lt {{ $.Values.healthcheck.readinessthreshold }} ]
             do
-                if [ -e tls.crl ] && [ -e root.crl ] && [ -e ssl.crl ] && [ -e ../done.txt ]
+                if [ -e done.txt ]
                 then
-                  echo "found crl certificates, performing vault put"
+                  mkdir ${BASE_DIR}/DATA/root
+                  cd ${BASE_DIR}/DATA/root
+                  cp ${BASE_DIR}/DATA/key-stores/root-key-store.jks ${BASE_DIR}/DATA/trust-stores/corda-ssl-trust-store.jks ${BASE_DIR}/DATA/trust-stores/network-root-truststore.jks ${BASE_DIR}/DATA/key-stores/corda-ssl-root-keys.jks ${BASE_DIR}/DATA/key-stores/tls-crl-signer-key-store.jks ${BASE_DIR}/DATA/key-stores/subordinate-key-store.jks ${BASE_DIR}/DATA/root;
```

Comment:
Why not change the pki.conf instead to generate the jks files in correct folders as you want.

Reply:
All the keystores are generated inside key-stores folder and all the truststores are generated inside trust-stores folder. And when we need some specific one at different path I am just putting that one at that path to match the path of vault. Isn't it more logical? what do u say?

Reply:
The logical one should be to update the conf file so that the files are generated in the custom folders from start. Instead of moving around by a "cp" command